### DISCUSSION TITLE
Using HTTP.jl as a static file server

### DISCUSSION THREAD

**Discussion by phanos-engineer** [JuliaWeb/HTTP.jl#979] [Show and tell]
I couldn't find any information in the documents but I was able to figure out a way to server static files. This allows you to serve a directory as a simple web server.

Assume you have these files to serve

```
www/index.html
www/assets/img/logo.png
www/assets/js/mystuff.js
www/assets/img/svg/picture1.jpg
```

So you have 4 directory depth. You can use `HTTP.serve` to serve your pages

```julia
    
    using HTTP,JSON3,Sockets;
    using Base.Filesystem
    const ROUTER = HTTP.Router()

    #four directory deep static stuff being hosted
    HTTP.register!(ROUTER, "GET", "", static_pages)
    HTTP.register!(ROUTER, "GET", "/*", static_pages)
    HTTP.register!(ROUTER, "GET", "/*/*", static_pages)
    HTTP.register!(ROUTER, "GET", "/*/*/*", static_pages)
    HTTP.register!(ROUTER, "GET", "/*/*/*/*", static_pages)
    HTTP.register!(ROUTER, "GET", "/*/*/*/*/*", static_pages)


    @info "Starting the server"
    HTTP.serve(ROUTER, Sockets.localhost, 8888 )

```

where the function `static_pages` is defined as

```julia
basepath="/var/myproject/"
function static_pages(req::HTTP.Request)
    ...
    relative_path=basepath*req.target
    body=read( relative_path )
    return HTTP.Response( 200, headers , body)
end
```
This will work but **shouldn't** be used for a production level server.

**Comment by quinnj** [comment]
Related issue: https://github.com/JuliaWeb/HTTP.jl/issues/567

**Comment by likev** [comment]
`HTTP.register!(ROUTER, "GET", "/**", static_pages)` should work
see https://juliaweb.github.io/HTTP.jl/stable/reference/#HTTP.Handlers.register!
